resume-buddy/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,118 @@
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    comments = new ArrayList<>();
+    String reviewer, reviewee, type, text, id;
+    Date date;
+    for (Entity entity : results.asIterable()) {
+      try {
+        reviewer = (String) entity.getProperty("reviewer");
+        reviewee = (String) entity.getProperty("reviewee");
+        type = (String) entity.getProperty("type");
+        text = (String) entity.getProperty("text");
+        date = (Date) entity.getProperty("date");
+        id = (String) entity.getProperty("uuid");
+      } catch (ClassCastException e) {
+        System.err.println("Could not cast entry property");
+        break;
+      }
+
+      comments.add(new Comment(reviewer, reviewee, text, type, date, id));
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String type = getParameter(request, "type", "");
+    String text = getParameter(request, "text", "");
+    Date date = new Date();
+
+    // TODO: Get real reviewer and reviewee info when auth implemented
+    String reviewer = "";
+    String reviewee = "";
+
+    UUID id = UUID.randomUUID();
+    while (collides(id)) {
+      id = UUID.randomUUID();
+    }
+
+    Entity commentEntity = new Entity("Review-comments");
+    commentEntity.setProperty("reviewer", reviewer);
+    commentEntity.setProperty("reviewee", reviewee);
+    commentEntity.setProperty("type", type);
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("date", date);
+    commentEntity.setProperty("uuid", id.toString());
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/resume-review.html");
+  }
+
+  /** Gets parameter from the list and changes the value by default if empty */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /** Checks if the id collides with other ids in datastore */
+  private boolean collides(UUID id) {
+    Query query = new Query("Review-comments");
+
+    Filter uuidPropertyFilter = new FilterPredicate("uuid", FilterOperator.EQUAL, id.toString());
+    query.setFilter(uuidPropertyFilter);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      String entityId = (String) entity.getProperty("uuid");
+      if (entityId.equals(id.toString())) {
+        return true;
+      }
+    }
+
+    return false;
```

Comment:
I think the `uuidPropertyFilter` should take care of anything in `results` matching the id. So this block should be replaceable with `return results.countEntities(FetchOptions.Builder.withDefaults()) != 0;`